docs/proposals/multiple-sources-for-applications.md

```diff
@@ -0,0 +1,231 @@
+---
+title: Neat-enhancement-idea
+authors:
+  - "@ishitasequeira" # Authors' github accounts here.
+sponsors:
+  - TBD        # List all intereste parties here.
+reviewers:
+  - "@jannfis"
+  - "@crenshaw-dev"
+approvers:
+  - "@jannfis"
+  - TBD
+
+creation-date: 2022-01-28
+last-updated: 2022-01-28
+---
+
+# Multiple Sources for application
+
+Support more than one source for creating an Application.
+
+Related Issues: 
+* [Proposal: Support multiple sources for an application](https://github.com/argoproj/argo-cd/issues/677)
+* [Helm chart + values from Git](https://github.com/argoproj/argo-cd/issues/2789)
+
+## Open Questions
+* Adding external sources to the Application reource would add additional latencies for creation/reconcilation process. Should we add it to the doc in Risks?
+
+## Summary
+
+Currently, Argo CD supports Applications with only a single ApplicationSource. In certain scenarios, it would be useful to support more than one source for the application. For example, consider a need to create multiple applications with the same manifests. Today we would have to copy manifest files from one application to another.
+
+Creating applications from multiple sources would allow users to configure multiple services stored at various sources within the same application.
+
+## Motivation
+
+The main motivation behind this enhancement proposal is to allow users to create an application using services that are stored in various sources.
+
+### Goals
+
+The goals of the enhancement are:
+
+#### **Supporting multiple sources for creating an application**
+
+Users should be able to specify multiple sources to add services to the application. Argo CD should compile all the sources and reconcile each source individually for creating the application.
+
+#### **Allow specifying external value files for Helm repositories**
+
+Users should be able to specify different sources for Helm charts and values files. The Helm charts specified by the user could be available in Git or Helm repository and the value files are stored in Git. Argo CD should track changes in both the Helm charts and the value files repository and reconcile the application.
```

Comment:
The initial requirement is to be able to sync from a repo of values, which is different from the chart repo.
It does not necessarily mean that ArgoCD should sync from both chart and values sources.
In practice, the package version will most probably be explicit in the application.yaml.

Of course, the more the better, within acceptable performances impacts.